specs/core/0_beacon-chain.md

```diff
@@ -2065,11 +2068,12 @@ def process_penalties_and_exits(state: BeaconState) -> None:
 
     def eligible(index):
         validator = state.validator_registry[index]
-        if validator.penalized_epoch <= current_epoch:
-            penalized_withdrawal_epochs = LATEST_PENALIZED_EXIT_LENGTH // 2
-            return current_epoch >= validator.penalized_epoch + penalized_withdrawal_epochs
+        # Cannot exit if you already have
+        if validator.withdrawable_epoch < FAR_FUTURE_EPOCH:
+            return False
+        # Can exit if at least the minimum amount of time has passed
         else:
-            return current_epoch >= validator.exit_epoch + MIN_VALIDATOR_WITHDRAWAL_EPOCHS
+            return current_epoch >= validator.exit_epoch + MIN_VALIDATOR_WITHDRAWABILITY_DELAY
```

Comment:
As the code is written it seems that if `current_epoch >= validator.exit_epoch + MIN_VALIDATOR_WITHDRAWABILITY_DELAY` then necessarily `validator.withdrawable_epoch < FAR_FUTURE_EPOCH` so that `eligible` always returns False.

Reply:
The first condition is about `withdrawable_epoch` whereas the second is about `exit_epoch`.
@JustinDrake I think you were conflating these two variables, no?

Reply:
I was making another silly mistake 🤦‍♂️